### ISSUE TITLE
Keyboard shortcut to "Close and Halt"

### ISSUE THREAD

**Issue by AGS-Knight** [jupyter/notebook#1097]
I'm trying to create a keyboard shortcut to invoke the  `Close and Halt` dropdown button in an open notebook.  I've tried using the following without success:

```
%%javascript

Jupyter.keyboard_manager.command_shortcuts.add_shortcut('ctrl-q', {
    help : 'Halt and Close',
    help_index : 'zz',
    handler : function (event) {
        IPython.notebook.kill_and_exit();
        return false;
    }}
);
```

If I use another method, say `clear_all_output` in place of `kill_and_exit`, the shortcut works as expected.  Am I using the wrong method here?

**Comment by Carreau** [comment]
IIRC this is a browser limitation, you cannot trigger a window closing too long after an user action, which often need to be a click, or someone else. The definition of "Too long" is variable, and I would expect we do too many indirection and the request to close is cancelled. 

We can try to investigate.

**Comment by AGS-Knight** [comment]
Thanks @Carreau , this might not be a crucial feature, but would help workflow, especially without the ability to preview notebooks before opening them.

**Comment by jaypeedevlin** [comment]
+1 on this, it would assist workflow for when you're making changes to lots of notebooks in sequence.

**Comment by tnlin** [comment]
+1, if no one doing this, I can do it

**Comment by Carreau** [comment]
5.0 (now in rc2) have the ability to create custom shortcuts from the help menu. So there is no need for patch, for the shortcut, though it is not exposed as an action. It should not be too hard to do .

**Comment by takluyver** [comment]
I think the active part is to expose the function behind the 'close and halt' menu entry:

https://github.com/jupyter/notebook/blob/bdd62b3c58051cd9c12cd77f27c289b711ed0b29/notebook/static/notebook/js/menubar.js#L230-L240

As an [action](https://github.com/jupyter/notebook/blob/master/notebook/static/notebook/js/actions.js), so that users can easily assign it a keyboard shortcut in the UI.

**Comment by takluyver** [comment]
Fixed by #3314 - it doesn't add a shortcut by default, but it makes it an action so you can easily configure a shortcut in the notebook interface.
